eng/validation/remove-oldbranches.ps1

```diff
@@ -0,0 +1,98 @@
+$testRoot = Join-Path -Path $testRootBase -ChildPath $([System.IO.Path]::GetRandomFileName())
+New-Item -Path $testRoot -ItemType Directory | Out-Null
+
+function GitHub-Clone($repoName) 
+{
+    & git clone $global:githubUri $(Get-Repo-Location $repoName)
+    Push-Location -Path $(Get-Repo-Location $repoName)
+    & git config user.email "${global:githubUser}@test.com"
+    & git config user.name $global:githubUser
+    Pop-Location
+}
+
+function Get-Repo-Location($repoName){ "$testRoot\$repoName" }
+
+function Git-Command($repoName) {
+    Push-Location -Path $(Get-Repo-Location($repoName))
+    try {
+        $gitParams = $args
+        if ($gitParams.GetType().Name -ne "Object[]") {
+            $gitParams = $gitParams.ToString().Split(" ")
+        }
+        Write-Host "Running 'git $gitParams' from $(Get-Location)"
+        $commandOutput = & git @gitParams; if ($LASTEXITCODE -ne 0) { throw "Git exited with exit code: $LASTEXITCODE" } else { $commandOutput }
+        $commandOutput
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+## Global Variables
+$global:githubUri = "https://${global:githubUser}:${global:githubPAT}@github.com/${global:githubOrg}/${global:githubRepoName}"
+$global:azdoUri = "https://${global:githubUser}:${global:azdoToken}@dev.azure.com/${global:azdoOrg}/${global:azdoProject}/_git/${global:azdoRepoName}"
+$global:remoteName = ($global:azdoOrg + "-" + $global:azdoRepoName)
+$global:branchNamePrefix = "refs/heads/dev/" + $global:githubUser + "/arcade-"
+$global:darcAzDORepoName = "https://dev.azure.com/${global:azdoOrg}/${global:azdoProject}/_git/${global:azdoRepoName}"
+$global:lastBranch = $global:branchNamePrefix + $global:arcadeSdkVersion
+
+Write-Host "Cloning '${global:githubRepoName} from GitHub"
+GitHub-Clone $global:githubRepoName
+
+Write-Host "Add remote to Azure DevOps"
+Git-Command $global:githubRepoName remote add $remoteName $global:azdoUri
+
+$remoteBranches = @()
+Write-Host "Getting remote branches for '${global:githubRepoName}' on Azure DevOps"
+$remoteBranches = Git-Command $global:githubRepoName ls-remote --heads $global:azdoUri
+
+foreach($remoteBranch in $remoteBranches)
+{
+    ## Delete all of the old branches except for any that match the current version of Arcade being validated
+    if(($remoteBranch -match ".*($global:branchNamePrefix).*") -and -not ($remoteBranch -match ".*($global:lastBranch).*"))
+    {
+        $branchName = ($remoteBranch -split "`t")[1]
+        try
+        {
+            Write-Host "Delete default channel and branch for branch named '${branchName}'"
+            & darc delete-default-channel --channel "General Testing" --branch $branchName --repo $global:darcAzDORepoName --azdev-pat $global:azdoToken --password $global:bartoken
+            Git-Command $global:githubRepoName push $remoteName --delete $branchName
+        }
+        catch
+        {
+            Write-Warning "Unable to delete default channel or branch when cleaning up"
```

Comment:
Is usually helpful to say what value caused the exception to be triggered. In this case what was the branch name and repo for instance. Also, maybe you can print "$_" - it may provide some useful info during an investigation?!

Reply:
Removed this part of the code, but still ensured that any other messages contained more information for debugging purposes.